homeassistant/components/withings/__init__.py

```diff
@@ -84,29 +85,41 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Withings from a config entry."""
-    # Normalize the config entry.
-    webhook_id = entry.data.get(CONF_WEBHOOK_ID, webhook.async_generate_id())
-    hass.config_entries.async_update_entry(
-        entry,
-        unique_id=entry.data["token"]["userid"],  # Upgrade existing configs.
-        data={
-            **entry.data,
-            **{
-                const.CLIENT_ID: hass.data[DOMAIN][const.CONFIG][const.CLIENT_ID],
-                const.CLIENT_SECRET: hass.data[DOMAIN][const.CONFIG][
-                    const.CLIENT_SECRET
-                ],
-                const.USE_WEBHOOK: hass.data[DOMAIN][const.CONFIG][const.USE_WEBHOOK],
-                CONF_WEBHOOK_ID: webhook_id,
-                const.CONF_WEBHOOK_URL: entry.data.get(
-                    const.CONF_WEBHOOK_URL, webhook.async_generate_url(hass, webhook_id)
-                ),
+    # Add a unique id if it's an older config entry.
+    if entry.unique_id != entry.data["token"]["userid"]:
+        hass.config_entries.async_update_entry(
```

Comment:
Instead of updating twice, keep track of an `updates` dict. 

Then after checks:

```python
if updates:
    hass.config_entries.async_update_entry(entry, **updates)
```